tests/integration/test.replication_events.js

```diff
@@ -170,5 +170,71 @@ adapters.forEach(function (adapters) {
       }).catch(done);
     });
 
+
+    // this test sets up a 2 way replication which initially transfers
+    // documents from a remote to a local database. Once the initial
+    // sync is complete, it inserts documents locally - the changes
+    // should propagate to the remote database and then back to the
+    // local database via the live replications. However, we do not
+    // expect any "change" events to fire for these documents since
+    // they already exist locally.
+    it('#4627 Test no duplicate changes in live replication', function () {
+      return new PouchDB.utils.Promise(function (resolve, reject) {
```

Comment:
Replication tests can very rarely use promises, in the other patch @nolanlawson was mainly referring to `db.post(obj, function`, for this test it looks cleaner to use `done()` instead of returning a promise